WireAnalytics/Tests/WireAnalyticsTests/AnalyticsManagerTests.swift

```diff
@@ -0,0 +1,139 @@
+//
+// Wire
+// Copyright (C) 2024 Wire Swiss GmbH
+//
+// This program is free software: you can redistribute it and/or modify
+// it under the terms of the GNU General Public License as published by
+// the Free Software Foundation, either version 3 of the License, or
+// (at your option) any later version.
+//
+// This program is distributed in the hope that it will be useful,
+// but WITHOUT ANY WARRANTY; without even the implied warranty of
+// MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE. See the
+// GNU General Public License for more details.
+//
+// You should have received a copy of the GNU General Public License
+// along with this program. If not, see http://www.gnu.org/licenses/.
+//
+
+import XCTest
+
+@testable import WireAnalytics
+@testable import WireAnalyticsSupport
+
+class AnalyticsManagerTests: XCTestCase {
+
+    // MARK: - Properties
+
+    private var analyticsService: MockAnalyticsService!
+    private var sut: AnalyticsManager!
+
+    // MARK: - setUp
+
+    override func setUp() {
+        super.setUp()
+        analyticsService = MockAnalyticsService()
+
+        analyticsService.startAppKeyHost_MockMethod = { _, _ in }
+        analyticsService.beginSession_MockMethod = {}
+        analyticsService.endSession_MockMethod = {}
+        analyticsService.changeDeviceID_MockMethod = { _ in }
+        analyticsService.setUserValueForKey_MockMethod = { _, _ in }
+
+        sut = AnalyticsManager(
+            appKey: "testAppKey",
+            host: URL(string: "https://test.com")!,
+            analyticsService: analyticsService
+        )
+    }
+
+    // MARK: - tearDown
+
+    override func tearDown() {
+        analyticsService = nil
+        sut = nil
+
+        super.tearDown()
+    }
+
+    func testInitialization() {
+        XCTAssertEqual(analyticsService.startAppKeyHost_Invocations.count, 1)
+        guard analyticsService.startAppKeyHost_Invocations.count == 1 else {
+            XCTFail("Expected 1 invocation of startAppKeyHost, but got \(analyticsService.startAppKeyHost_Invocations.count)")
+            return
+        }
+        XCTAssertEqual(analyticsService.startAppKeyHost_Invocations[0].appKey, "testAppKey")
+        XCTAssertEqual(analyticsService.startAppKeyHost_Invocations[0].host, URL(string: "https://test.com")!)
+    }
+
+    func testSwitchUser() {
+        // GIVEN
+        let userProfile = AnalyticsUserProfile(
+            analyticsIdentifier: "testUser123",
+            teamInfo: .init(id: "team1", role: "admin", size: 10)
+        )
+
+        // WHEN
+        _ = sut.switchUser(userProfile)
+
+        // THEN
+        XCTAssertEqual(analyticsService.endSession_Invocations.count, 1)
+        XCTAssertEqual(analyticsService.changeDeviceID_Invocations.count, 1)
```

Comment:
This assertion is redundant because of the guard below.

Reply:
Done: https://github.com/wireapp/wire-ios/commit/73febf55738dcb9b1c56a1e3703c2cb786de9088